CHANGELOG/CHANGELOG-4.2.md

```diff
@@ -0,0 +1,244 @@
+# Release notes for v4.2.0
+
+[Documentation](https://kubernetes-csi.github.io)
+
+# Changelog since v4.1.0
+
+## Changes by Kind
+
+### Feature
+
+- Updates Kubernetes dependencies to v1.22.0 ([#570](https://github.com/kubernetes-csi/external-snapshotter/pull/570), [@chrishenzie](https://github.com/chrishenzie)) [SIG Storage]
+- Updates csi-lib-utils dependency to v0.10.0 ([#574](https://github.com/kubernetes-csi/external-snapshotter/pull/574), [@chrishenzie](https://github.com/chrishenzie))
```

Comment:
I've been discussing thus with @chrishenzie in https://github.com/kubernetes-csi/external-attacher/pull/312#issuecomment-896219493. In a nutshell, I find it odd that a dependency update is listed as new feature when it doesn't actually lead to any user-visible change of behavior. If behavior changes, I would expect to see that called out.

For "normal" dependency updates that are just done to keep up-to-date or prepare for some future code change I find "cleanup" more appropriate.

Reply:
I also think it is odd to keep dependency update under feature.  I kept it here just because the original PR is labeled as feature.  I agree it is more appropriate to call these "cleanup".

Reply:
Moved under "cleanup".

Reply:
Thank you for the feedback @pohly, some of these dependency updates were listed under feature for the provisioner, I will move this under cleanup for future releases.